packages/cli/src/lib/version.ts

```diff
@@ -60,3 +62,22 @@ export function findVersion() {
 
 export const version = findVersion();
 export const isDev = fs.pathExistsSync(paths.resolveOwn('src'));
+
+export function createPackageVersionProvider(lockfile?: Lockfile) {
+  return (name: string, versionHint?: string) => {
+    const targetVersion = versionHint || packageVersions[name];
+    if (!targetVersion) {
+      throw new Error(`No version available for package ${name}`);
+    }
+
+    const lockfileEntries = lockfile?.get(name);
+    if (lockfileEntries?.some(entry => entry.range === '*')) {
+      return '*';
```

Comment:
hm! i would have expected this one to not be "greedy", but rather as a last resort if no actual semver ranges were to be found 🤔

Reply:
The reasoning here is that if there's a `*` it's usually for good reason. Might not always be true though and there might just be some dependency somewhere that has it unnecessarily. I'm thinking we could do this for `@types/*` packages, but not the rest